src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/RazorDefinitionProvider.ts

```diff
@@ -30,16 +31,28 @@ export class RazorDefinitionProvider
             // C# knows about line pragma, if we're getting a direction to a virtual c# document
             // that means the piece we're trying to navigate to does not have a representation in the
             // top level file.
-            definitions = (definitions as vscode.Location[]).filter(definition => {
-                return !(definition.uri.path.endsWith(virtualCSharpSuffix));
-            });
+            for (const definition of definitions) {
+                const uriPath = getUriPath(definition.uri);
+                if (uriPath.endsWith(virtualCSharpSuffix)) {
```

Comment:
Need to also handle the bg path:  https://github.com/aspnet/AspNetCore-Tooling/blob/ec2727d7c06c60361f70730eb4aa1d78d4543fde/src/Razor/src/Microsoft.AspNetCore.Razor.OmniSharpPlugin/BackgroundDocumentProcessedPublisher.cs#L36

Reply:
Should be handled just below this, since `__virtual` is still at the end we just have to remove `__bg.__virtual` AND `__virtual` right?

Reply:
The way you have it in the PR should work